test-it/ExpectedResults/VariantAttributes.vtr

```diff
@@ -1,30 +1,30 @@
 TestSetVariantAttributes()
-Set: myVar=^Variant, replaced=false error=(false 0)
-Set: myVar=^Variant, replaced=true error=(false 0)
-Set: myVar=^Variant, replaced=false error=(false 0)
-Set: myVar=^Variant, replaced=true error=(false 0)
+Set: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), replaced=false error=(false 0)
+Set: myVar=(_data:255 _attributes:(aaa:(_data:yyy _attributes:null))), replaced=true error=(false 0)
+Set: myVar=(_data:255 _attributes:(aaa:(_data:yyy _attributes:null) bbb:(_data:yyy _attributes:null))), replaced=false error=(false 0)
+Set: myVar=(_data:255 _attributes:(aaa:(_data:yyy _attributes:null) bbb:(_data:zzz _attributes:null))), replaced=true error=(false 0)
 
 TestGetVariantAttributes()
-Set: myVar=^Variant, replaced=false error=(false 0)
-Get: myVar=^Variant, found=true myStr=xxx error=(false 0)
-Get: myVar=^Variant, found=false myInt=255 error=(true 91)
-Get: myVar=^Variant, found=false myInt=255 error=(false 0)
+Set: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), replaced=false error=(false 0)
+Get: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), found=true myStr=xxx error=(false 0)
+Get: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), found=false myInt=255 error=(true 91)
+Get: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), found=false myInt=255 error=(false 0)
 
 TestGetVariantAttributesAll()
-Get: myVar=^Variant, error=(false 0) attribNames=()
-Set: myVar=^Variant, replaced=false error=(false 0)
-Get: myVar=^Variant, error=(false 0) attribNames=('aaa') values=(xxx)
-Set: myVar=^Variant, replaced=false error=(false 0)
-Get: myVar=^Variant, error=(false 0) attribNames=('aaa' 'bbb') values=(xxx 255)
+Get: myVar=(_data:255 _attributes:null), error=(false 0) attribNames=()
+Set: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), replaced=false error=(false 0)
+Get: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), error=(false 0) attribNames=('aaa') values=(xxx)
+Set: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null) bbb:(_data:255 _attributes:null))), replaced=false error=(false 0)
+Get: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null) bbb:(_data:255 _attributes:null))), error=(false 0) attribNames=('aaa' 'bbb') values=(xxx 255)
 
 TestDeleteVariantAttributes()
-Set: myVar=^Variant, replaced=false error=(false 0)
-Delete: myVar=^Variant, found=false error=(false 0)
-Get: myVar=^Variant, found=true error=(false 0) values=(xxx)
-Delete: myVar=^Variant, found=true error=(false 0)
-Get: myVar=^Variant, found=false error=(false 0) values=(z)
-Set: myVar=^Variant, replaced=false error=(false 0)
-Set: myVar=^Variant, replaced=false error=(false 0)
-Delete: myVar=^Variant, found=true error=(false 0)
-Get: myVar=^Variant, found=(false false) error=(false 0) values=(z 0)
+Set: myVar=(_data:255 _attributes:(aaa:(_data:xxx _attributes:null))), replaced=false error=(false 0)
```

Comment:
Values of data types used in variant (as data or attribute name or attribute value) should be formatted same as if they were used directly for the data type. For example, strings should be single quoted. Here "aaa" and "xxx" are printed without quotes.  It is likely because we are formatting in the new function directly as opposed to calling "Format" to format the string.

Reply:
Updated this. I wasn't setting a flag correctly.